quantum/rgb_matrix/animations/raindrops_anim.h

```diff
@@ -2,35 +2,40 @@
 RGB_MATRIX_EFFECT(RAINDROPS)
 #    ifdef RGB_MATRIX_CUSTOM_EFFECT_IMPLS
 
-static void raindrops_set_color(int i, effect_params_t* params) {
-    if (!HAS_ANY_FLAGS(g_led_config.flags[i], params->flags)) return;
-    hsv_t hsv = {0, rgb_matrix_config.hsv.s, rgb_matrix_config.hsv.v};
-
+static rgb_t raindrops_set_color(hsv_t hsv) {
     // Take the shortest path between hues
-    int16_t deltaH = ((rgb_matrix_config.hsv.h + 180) % 360 - rgb_matrix_config.hsv.h) / 4;
+    int16_t deltaH = ((hsv.h + 180) % 360 - hsv.h) / 4;
     if (deltaH > 127) {
         deltaH -= 256;
     } else if (deltaH < -127) {
         deltaH += 256;
     }
 
-    hsv.h     = rgb_matrix_config.hsv.h + (deltaH * (random8() & 0x03));
-    rgb_t rgb = rgb_matrix_hsv_to_rgb(hsv);
-    rgb_matrix_set_color(i, rgb.r, rgb.g, rgb.b);
+    hsv.h += (deltaH * random8_max(3));
+    return rgb_matrix_hsv_to_rgb(hsv);
 }
 
 bool RAINDROPS(effect_params_t* params) {
+    static uint16_t idx = 0;
+    static rgb_t    rgb = {0};
+
+    if (scale16by8(g_rgb_timer, qadd8(rgb_matrix_config.speed, 16)) % 10 == 0) {
```

Comment:
This should be done only on the first iteration of the frame rendering (only the actual setting of the chosen LED needs to be delayed until the appropriate iteration).
```suggestion
    if ((params->iter == 0) && (scale16by8(g_rgb_timer, qadd8(rgb_matrix_config.speed, 16)) % 10 == 0)) {
```

Reply:
Committed suggestion.